core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala

```diff
@@ -548,6 +548,7 @@ object JarCreationTest extends Logging {
     if (result.nonEmpty) {
       throw new Exception("Could not load user class from jar:\n" + result(0))
     }
+    sc.stop()
```

Comment:
Looks like there is something wrong with the shutdown process of local-cluster. There are several rounds of killing executors and launching new executors before `Successfully stopped SparkContext` shows up. You can take a look at  https://issues.apache.org/jira/browse/SPARK-8643 for the log I got.